packages/elements/src/components/ino-nav-drawer/ino-nav-drawer.tsx

```diff
@@ -67,9 +66,21 @@ export class NavDrawer implements ComponentInterface {
   @Prop() a11yLabels?: NavDrawerLabels = {
     content: 'Main Navigation',
     footer: 'Footer Navigation',
-    toggleBtn: 'Toggle Navigation'
+    toggleBtn: 'Toggle Navigation',
   };
 
+  /**
+   * If true, the mobile drawer will be shown. Automatically sets the variant to `modal`.
+   */
+  @Prop() isMobile?: boolean = false;
+
+  // set the variant based on the isMobile prop
+  get programmaticVariant(): NavDrawerVariant {
+    if (this.isMobile) {
+      return 'modal';
+    }
+    return this.variant;
+  }
```

Comment:
You can use stencils `@Watch()` annotation to detect changes in `isMobile`. This function can then set the variant depending on the `isMobile` content.

```
  @Watch('isMobile')
  handleIsMobileChange() {
    if (this.isMobile) {
      this.variant = 'modal';
    }
  }
````
Further down the line, you only have to check `this.variant` to get according css classes